CONTRIBUTING.md

```diff
@@ -4,4 +4,64 @@ Thank you for your interest in the Langium project! The following is a set of gu
 
 ## Prerequisites
 
-Developing Langium only requires `npm@^7.7.0`. For this you will need `Node.js 10` or greater.
\ No newline at end of file
+Developing Langium only requires `npm@^7.7.0`. For this you will need `Node.js 10` or greater.
+
+## Best practices for local development
+
+Langium itself contains a couple of language examples. If you want to write your own grammar you should use `yo langium` and create an independent project. If you just want to use Langium out of the box, then you are ready to start and don't need to necessarily read further. But if for whatever reason you want to extend Langium locally and you want to use your changes in your own language project then you should follow some recommendations.
+
+### npm dependency resolution pitfalls
+
+When you add a dependency to a `package.json` npm resolves this from the configured registry (usually https://registry.npmjs.org/). In other dependency management system you can create something like a dev or snapshot version and install it locally, but this is not possible as such with npm. If you want to achieve a similar behavior you need to link your packages globally (global refers to you user's account scope, local refers to the project scope). When you do this you forcefully overwrite what was downloaded from the npm registry. Incrementing the version of Langium itself will led to problems, because npm will not be able to resolve those from its registry and therefore fail any install attempt in projects referring to Langium.
+
+Langium is an npm workspace project with currently four main packages (`langium`, `langium-cli`, `langium-vscode` and `generator-langium`). If you execute `npm install` on the top-level of the project all dependencies will be gathered locally inside `node_modules` and all four packages or sub-projects will be build. **Warning**: Do not run npm install directly inside the four Langium packages as it will mess up dependency resolution once you link packages globally.
```

Comment:
```suggestion
Langium is an npm workspace project with currently four main packages (`langium`, `langium-cli`, `langium-vscode` and `generator-langium`). If you execute `npm install` on the top-level of the project all dependencies will be gathered locally inside `node_modules` and all four packages or sub-projects will be built. **Warning**: Do not run npm install directly inside the four Langium packages as it will mess up dependency resolution once you link packages globally.
```

Reply:
I have already adjusted the test locally and it will be part of an upcoming commit.